autoload/vimteractive.vim

```diff
@@ -1,9 +1,57 @@
+let s:iconn_cw_winnr = -1
+
+function! s:show_term(bufnr)
+	split
+	execute ":b " . g:vimteractive_terminal[a:bufnr]
+	wincmd p
+endfunction
+
+function! s:check_alive(bufnr)
```

Comment:
This function seems to both check if a terminal exists, and clear variables out if it doesn't (e.g. if the user has deleted it). Could any confusion arise for future extensions from this dual behaviour (and the fact that unused terminal buffers are only cleaned when this function is called)? Could you also add a comment line for this function?

Reply:
> Could any confusion arise for future extensions from this dual behaviour 

I don't think so. This function cleanups only buffers which obviously doesn't exist anymore.
> and the fact that unused terminal buffers are only cleaned when this function is called

This is bad, but I'm not sure how to do this other way. Autocommands are _strange_ when dealing with terminal buffers.

Reply:
How necessary is the ``vimteractive_current_term_type`` dictionary? I don't find this mapping that natural, since the 'terminal type' should be owned by the terminal rather than the buffer. Changing to a dictionary that maps terminals to terminal types might reduce the risk of miscleaning

Reply:
(or simply inferring the terminal type from the buffer name)

Reply:
Upon further reflection, one could possibly replace the dictionary mapping buffers to terminals with buffer-scoped variables (e.g. ``b:vimteractive_terminal``). This would obviate the need for cleaning, as they would go out of scope once the buffer was destroyed.

Reply:
I agree that it would be cooler, but can't understand how to get terms list without it.

Reply:
Perhaps could extract them using some kind of regex:

https://stackoverflow.com/questions/29803385/show-vim-buffers-matching-a-pattern/29803658

in the similar way to the manner that you currently create the list of ``term_type``s.

Reply:
There's a difference between regex matching against buffer we know is ours and against any buffer
User can be editing similar called file. We really shouldn't rely on buffer names.

Reply:
Agreed. You could possibly use some of the ideas in this code example:

http://www.giocc.com/track-active-terminal-buffers-with-vimscript.html

I think that most of that is designed to teach people how to write a package, but there are some good ideas using autocommands to update a list of buffers of interest.